### DISCUSSION TITLE
Revisions differences

### DISCUSSION THREAD

**Discussion by djordjedjukic** [ravendb/ravendb#12166] [Q&A] [answered]
Is there a way to get differences between 2 revisions like you get with WhatChanged() method?

**Comment by ayende** [comment, marked as answer]
There is no API for that. The Studio is able to do that, but it is doing this inline. By fetching the two revisions and comparing their texts.

I would recommend that you take the same approach here. Just fetch the two documents, output their json with formatting enabled and then use any text diff tooling.

Alternatively, you can get the two revisions and pass them to a library that would compare them.
An example of such a library is here:
https://github.com/GregFinzer/Compare-Net-Objects

That is just the first I found, no idea about its quality.

**Comment by DejanMilicic** [comment]
You can also leverage session's tracking of changes

```
        public void ExtractChanges()
        {
            using var session = DocumentStoreHolder.Store.OpenSession();

            Order order = session.Load<Order>("orders/823-A");
            Order revision = session.Advanced.Revisions.GetFor<Order>("orders/823-A").Skip(2).First();

            JsonConvert.PopulateObject(JsonConvert.SerializeObject(revision), order,
                new JsonSerializerSettings { ObjectCreationHandling = ObjectCreationHandling.Replace });

            var changes = session.Advanced.WhatChanged();
        }
```
